integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/AbstractIntegrationTest.java

```diff
@@ -80,21 +83,51 @@ public void beforeSuite(ITestContext testngContext) {
     }
 
     @BeforeMethod
-    public void beforeMethod(Method method) {
+    public void beforeTest(Method method, Object[] params) {
         MDC.put("testlabel", method.getDeclaringClass().getSimpleName() + '.' + method.getName());
+        collectTestCaseDescription(method, params);
+    }
+
+    private TestCaseDescription collectTestCaseDescription(Method method, Object[] params) {
+        Description declaredAnnotation = method.getDeclaredAnnotation(Description.class);
+        TestCaseDescription testCaseDescription = null;
+        if (declaredAnnotation != null) {
+            testCaseDescription = new TestCaseDescriptionBuilder()
+                    .given(declaredAnnotation.given())
+                    .when(declaredAnnotation.when())
+                    .then(declaredAnnotation.then());
+            ((TestContext) params[0]).withDescription(testCaseDescription);
+        }
+        if (testCaseDescription == null) {
+            Parameter[] parameters = method.getParameters();
+            for (int i = 1; i < parameters.length; i++) {
+                if (parameters[i].getAnnotation(Description.class) != null) {
+                    Object param = params[i];
+                    if (param instanceof TestCaseDescription) {
+                        testCaseDescription = (TestCaseDescription) param;
+                        ((TestContext) params[0]).withDescription(testCaseDescription);
```

Comment:
Is not clear to me, are we looking for the first element? Why not exit if so. Or If we are looking for all elements why we always change `params[0]`??